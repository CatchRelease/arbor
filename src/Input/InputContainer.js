import styled from '@emotion/styled';
import PropTypes from 'prop-types';

const InputContainer = styled.div`
  margin-bottom: ${props =>
    props.caption ? props.theme.space.smaller : props.theme.space.regular};
`;

InputContainer.propTypes = {
  caption: PropTypes.string
};

InputContainer.defaultProps = {
  caption: ''
};

export default InputContainer;
