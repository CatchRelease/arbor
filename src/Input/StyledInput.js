import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import { placeholder, transparentize } from 'polished';
import { maxWidth } from 'styled-system';

const baseStyles = ({ theme }) => css`
  border-radius: ${theme.radii.small};
  border: ${theme.borderWidth.small} solid ${theme.colors.border.default};
  box-sizing: border-box;
  color: ${theme.colors.text.default};
  font-family: ${theme.brandFont};
  font-size: ${theme.fontSizes.input.default};
  line-height: ${theme.lineHeights.small};
  margin: ${theme.space.smallest} 0;
  padding: ${theme.space.smaller};
  width: 100%;

  @media (max-width: ${theme.breakpointsMap.md}) {
    font-size: ${theme.fontSizes.input.mobile};
  }

  ${placeholder({ color: theme.colors.grey60 })};

  &:focus {
    border: ${theme.borderWidth.small} solid ${theme.colors.brand.dark};
    box-shadow: 0 0 8px ${transparentize(0.9, theme.colors.brand.light)};
    outline: none;
  }

  &:disabled {
    background: ${theme.colors.background.default};
  }

  &:required {
    box-shadow: none;
  }
`;

const largeStyles = ({ theme, large }) =>
  large &&
  css`
    padding: ${theme.space.small} ${theme.space.smaller};
  `;

const StyledInput = styled.input`
  ${baseStyles};
  ${largeStyles};
  ${maxWidth};
`;

StyledInput.propTypes = {
  large: PropTypes.bool,
  ...maxWidth.propTypes
};

StyledInput.defaultProps = {
  as: 'input',
  large: false
};

export default StyledInput;
