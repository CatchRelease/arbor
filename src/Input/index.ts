export { default, Props as InputProps } from './Input';
export { default as inputStyles } from './inputStyles';
