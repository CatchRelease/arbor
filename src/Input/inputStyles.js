import { css } from '@emotion/react';
import { transparentize } from 'polished';

const disabledStyles = (theme) => css`
  background: ${theme.colors.background.default};
`;

const focusedStyles = (theme) => css`
  border: ${theme.borderWidth.small} solid ${theme.colors.intent.brand.dark};
  box-shadow: 0 0 8px ${transparentize(0.9, theme.colors.intent.brand.light)};
  outline: none;
`;

const invalidStyles = (theme) => css`
  border: ${theme.borderWidth.small} solid ${theme.colors.palette.red.default};
  background: ${theme.colors.palette.red.lighter};
`;

/* stylelint-disable property-no-vendor-prefix */
const hideArrowsAddedByBrowsers = css`
  ::-webkit-outer-spin-button,
  ::-webkit-inner-spin-button {
    -webkit-appearance: none;
    margin: 0;
  }
  -moz-appearance: textfield;
`;
/* stylelint-enable property-no-vendor-prefix */

const inputStyles = ({ isDisabled, isFocused, isInvalid, theme }) => css`
  border-radius: ${theme.radii.small};
  border: ${theme.borderWidth.small} solid ${theme.colors.border.default};
  box-sizing: border-box;
  color: ${theme.colors.text.default};
  font-family: ${theme.brandFont};
  font-size: ${theme.fontSizes.input.default};
  line-height: ${theme.lineHeights.small};
  margin: ${theme.space.smallest} 0;
  padding: ${theme.space.smaller};
  width: 100%;

  @media (max-width: ${theme.breakpointsMap.md}) {
    font-size: ${theme.fontSizes.input.mobile};
  }

  ::placeholder {
    color: ${theme.colors.text.placeholder};
  }
  ${isDisabled ? disabledStyles(theme) : ''};
  ${isFocused ? focusedStyles(theme) : ''};
  ${isInvalid ? invalidStyles(theme) : ''};

  &:focus {
    ${focusedStyles(theme)};
  }

  &:disabled {
    ${disabledStyles(theme)};
  }

  &:required {
    box-shadow: none;
  }

  ${hideArrowsAddedByBrowsers}
`;

export default inputStyles;
