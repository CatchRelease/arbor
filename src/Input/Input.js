import React from 'react';
import PropTypes from 'prop-types';

import FormField from '../FormField';
import StyledInput from './StyledInput';

const Input = React.forwardRef(
  ({ caption, label, labelAside, id, ...props }, ref) => (
    <FormField caption={caption} id={id} label={label} labelAside={labelAside}>
      <StyledInput {...{ ...props, id, ref }} />
    </FormField>
  )
);

Input.propTypes = {
  /**
   * The actualy HTML element tag to use when rendering.
   */
  as: PropTypes.string,

  /**
   * Optional text to be displayed underneath the input.
   */
  caption: PropTypes.string,

  /**
   * HTML ID for the input to be used in conjunction with the label for
   * accessiblity.
   */
  id: PropTypes.string.isRequired,

  /**
   * Label text for the form field
   */
  label: PropTypes.string,

  /**
   * Component that goes to the right of the label. Does not have to only be text.
   */
  labelAside: PropTypes.node,

  /**
   * HTML Input type
   */
  type: PropTypes.string
};

Input.defaultProps = {
  as: 'input',
  caption: '',
  label: '',
  labelAside: null,
  type: 'text'
};

export default Input;
