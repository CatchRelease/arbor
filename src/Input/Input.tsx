import {
  ChangeEventHandler,
  FC,
  FocusEventHandler,
  forwardRef,
  InputHTMLAttributes,
  ReactNode,
  useState
} from 'react';

import FormField from '../FormField';
import StyledInput, { Props as StyledInputProps } from './StyledInput';

type Props = InputHTMLAttributes<HTMLInputElement> &
  StyledInputProps & {
    caption?: string | null;
    id: string;
    label?: string;
    labelAside?: ReactNode;
    onBlur?: FocusEventHandler<HTMLInputElement>;
    onChange?: ChangeEventHandler<HTMLInputElement>;
    type?: string;
    validate?: (value?: string | number | readonly string[]) => string | null;
  };

const Input: FC<Props> = forwardRef<HTMLInputElement, Props>(
  (
    {
      caption = '',
      label = '',
      labelAside = null,
      id,
      onBlur,
      onChange,
      type = 'text',
      validate = () => undefined,
      value: propValue,
      ...props
    },
    ref
  ) => {
    const [touched, setTouched] = useState(false);
    const [value, setValue] = useState(propValue);

    const errorMessage = touched && validate(value);

    const handleBlur: FocusEventHandler<HTMLInputElement> = (event) => {
      setTouched(true);

      if (onBlur) {
        onBlur(event);
      }
    };

    const handleChange: ChangeEventHandler<HTMLInputElement> = (event) => {
      setValue(event.target.value);

      if (onChange) {
        onChange(event);
      }
    };

    return (
      <FormField
        caption={errorMessage || caption}
        id={id}
        label={label}
        labelAside={labelAside}
      >
        <StyledInput
          {...{ ...props, id, ref, type }}
          value={propValue ? propValue.toString() : propValue}
          isInvalid={!!errorMessage}
          onBlur={handleBlur}
          onChange={handleChange}
        />
      </FormField>
    );
  }
);

export default Input;
