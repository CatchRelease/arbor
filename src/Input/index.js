import React from 'react';
import PropTypes from 'prop-types';

import InputContainer from './InputContainer';
import StyledInput from './StyledInput';
import Label from '../Label';
import Paragraph from '../Paragraph';

const Input = ({ caption, label, id, ...props }) => (
  <InputContainer caption={caption}>
    <Label htmlFor={id}>{label}</Label>
    <StyledInput {...{ ...props, id }} />
    {caption && (
      <Paragraph.span variant="tiny" color="red">
        {caption}
      </Paragraph.span>
    )}
  </InputContainer>
);

Input.propTypes = {
  caption: PropTypes.string,
  id: PropTypes.string.isRequired,
  large: PropTypes.bool
};

Input.defaultProps = {
  caption: '',
  large: false
};

export default Input;
