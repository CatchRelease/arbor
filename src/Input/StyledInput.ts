import { css, Theme } from '@emotion/react';
import styled from '@emotion/styled';
import { display, DisplayProps, maxWidth, MaxWidthProps } from 'styled-system';
import inputStyles, { InputStylesProps } from './inputStyles';

const largeStyles = ({ theme, large }: { theme: Theme; large?: boolean }) =>
  large &&
  css`
    padding: ${theme.space.small} ${theme.space.smaller};
  `;

export type Props = DisplayProps &
  InputStylesProps &
  MaxWidthProps & {
    large?: boolean;
  };

const StyledInput = styled.input<Props>`
  ${inputStyles};
  ${display};
  ${largeStyles};
  ${maxWidth};
`;

export default StyledInput;
