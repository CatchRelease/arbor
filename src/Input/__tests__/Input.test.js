import { ThemeProvider } from '@emotion/react';
import { mount } from 'enzyme';

import createWithTheme from '../../../utils/createWithTheme';
import theme from '../../theme';
import FormField from '../../FormField';
import Input from '../Input';
import StyledInput from '../StyledInput';

describe('<Input />', () => {
  it('renders an Input correctly', () => {
    const tree = createWithTheme(<Input id="example" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders a large Input correctly', () => {
    const tree = createWithTheme(<Input large id="example" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an Input with a caption correctly', () => {
    const tree = createWithTheme(
      <Input id="example" caption="I am a caption" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an Input with a label correctly', () => {
    const tree = createWithTheme(
      <Input label="My Input" id="example" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders as Input with a labelAside correctly', () => {
    const tree = createWithTheme(
      <Input labelAside={<div>sup?</div>} id="example" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders an Input with a type correctly', () => {
    const tree = createWithTheme(<Input type="number" id="example" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  context('with a validate function', () => {
    const render = (props) =>
      mount(
        <ThemeProvider theme={theme}>
          <Input
            caption="The caption"
            id="example"
            validate={(value) => (value === '42' ? null : 'Value must be 42')}
            value="42"
            {...props}
          />
        </ThemeProvider>
      );

    it('displays the caption when valid', () => {
      const wrapper = render();

      expect(wrapper.find(FormField)).toHaveProp('caption', 'The caption');
    });

    it('displays the caption when valid and blurred', () => {
      const wrapper = render();

      wrapper.find(StyledInput).simulate('blur');

      expect(wrapper.find(FormField)).toHaveProp('caption', 'The caption');
    });

    it('displays the caption when invalid but not yet blurred', () => {
      const wrapper = render();
      const input = wrapper.find(StyledInput);

      input.simulate('change', { target: { value: '123' } });

      expect(wrapper.find(FormField)).toHaveProp('caption', 'The caption');
    });

    it('displays the error message when invalid and blurred', () => {
      const wrapper = render();
      const input = wrapper.find(StyledInput);

      input.simulate('change', { target: { value: '123' } });
      input.simulate('blur');

      expect(wrapper.find(FormField)).toHaveProp('caption', 'Value must be 42');
    });

    it('marks the input valid when valid', () => {
      const wrapper = render();

      expect(wrapper.find(StyledInput)).toHaveProp('isInvalid', false);
    });

    it('marks the input invalid when invalid', () => {
      const wrapper = render();
      const input = wrapper.find(StyledInput);

      input.simulate('change', { target: { value: '123' } });
      input.simulate('blur');

      expect(wrapper.find(StyledInput)).toHaveProp('isInvalid', true);
    });

    it('calls onBlur with the original event', () => {
      const onBlur = jest.fn();
      const wrapper = render({ onBlur });

      const fakeEvent = { id: 'fake-event' };

      wrapper.find(StyledInput).simulate('blur', fakeEvent);

      expect(onBlur).toHaveBeenCalledWith(expect.objectContaining(fakeEvent));
    });

    it('calls onChange', () => {
      const onChange = jest.fn();
      const wrapper = render({ onChange });

      const fakeEvent = { id: 'fake-event', target: { value: '42' } };

      wrapper.find(StyledInput).simulate('change', fakeEvent);

      expect(onChange).toHaveBeenCalledWith(expect.objectContaining(fakeEvent));
    });
  });
});
