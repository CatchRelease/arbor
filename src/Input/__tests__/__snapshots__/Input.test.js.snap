// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders a large Input correctly 1`] = `
.emotion-4 {
  margin-bottom: 1rem;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border-radius: 3px;
  border: 1px solid #E0E1E2;
  box-sizing: border-box;
  color: #25272D;
  font-family: Motiva Sans;
  font-size: 0.875rem;
  line-height: 1.25;
  margin: 0.25rem 0;
  padding: 0.5rem;
  width: 100%;
  padding: 0.75rem 0.5rem;
}

@media (max-width:768px) {
  .emotion-2 {
    font-size: 1rem;
  }
}

.emotion-2::-webkit-input-placeholder {
  color: #989A9C;
}

.emotion-2:-moz-placeholder {
  color: #989A9C;
}

.emotion-2::-moz-placeholder {
  color: #989A9C;
}

.emotion-2:-ms-input-placeholder {
  color: #989A9C;
}

.emotion-2:focus {
  border: 1px solid #081D17;
  box-shadow: 0 0 8px rgba(182,239,208,0.1);
  outline: none;
}

.emotion-2:disabled {
  background: #F2F2F2;
}

.emotion-2:required {
  box-shadow: none;
}

<div
  className="emotion-4 emotion-5"
>
  <div
    className="emotion-0 emotion-1"
  >
    
  </div>
  <input
    className="emotion-2 emotion-3"
    id="example"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
  
</div>
`;

exports[`<Input /> renders an Input correctly 1`] = `
.emotion-4 {
  margin-bottom: 1rem;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border-radius: 3px;
  border: 1px solid #E0E1E2;
  box-sizing: border-box;
  color: #25272D;
  font-family: Motiva Sans;
  font-size: 0.875rem;
  line-height: 1.25;
  margin: 0.25rem 0;
  padding: 0.5rem;
  width: 100%;
}

@media (max-width:768px) {
  .emotion-2 {
    font-size: 1rem;
  }
}

.emotion-2::-webkit-input-placeholder {
  color: #989A9C;
}

.emotion-2:-moz-placeholder {
  color: #989A9C;
}

.emotion-2::-moz-placeholder {
  color: #989A9C;
}

.emotion-2:-ms-input-placeholder {
  color: #989A9C;
}

.emotion-2:focus {
  border: 1px solid #081D17;
  box-shadow: 0 0 8px rgba(182,239,208,0.1);
  outline: none;
}

.emotion-2:disabled {
  background: #F2F2F2;
}

.emotion-2:required {
  box-shadow: none;
}

<div
  className="emotion-4 emotion-5"
>
  <div
    className="emotion-0 emotion-1"
  >
    
  </div>
  <input
    className="emotion-2 emotion-3"
    id="example"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
  
</div>
`;

exports[`<Input /> renders an Input with a caption correctly 1`] = `
.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border-radius: 3px;
  border: 1px solid #E0E1E2;
  box-sizing: border-box;
  color: #25272D;
  font-family: Motiva Sans;
  font-size: 0.875rem;
  line-height: 1.25;
  margin: 0.25rem 0;
  padding: 0.5rem;
  width: 100%;
}

@media (max-width:768px) {
  .emotion-2 {
    font-size: 1rem;
  }
}

.emotion-2::-webkit-input-placeholder {
  color: #989A9C;
}

.emotion-2:-moz-placeholder {
  color: #989A9C;
}

.emotion-2::-moz-placeholder {
  color: #989A9C;
}

.emotion-2:-ms-input-placeholder {
  color: #989A9C;
}

.emotion-2:focus {
  border: 1px solid #081D17;
  box-shadow: 0 0 8px rgba(182,239,208,0.1);
  outline: none;
}

.emotion-2:disabled {
  background: #F2F2F2;
}

.emotion-2:required {
  box-shadow: none;
}

.emotion-6 {
  margin-bottom: 0.5rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  color: #D93838;
  line-height: 1.5;
  font-size: 0.75rem;
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-0 emotion-1"
  >
    
  </div>
  <input
    className="emotion-2 emotion-3"
    id="example"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
  <span
    className="emotion-4 emotion-5"
    color="palette.red.default"
  >
    I am a caption
  </span>
</div>
`;

exports[`<Input /> renders an Input with a label correctly 1`] = `
.emotion-8 {
  margin-bottom: 1rem;
}

.emotion-4 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-6 {
  border-radius: 3px;
  border: 1px solid #E0E1E2;
  box-sizing: border-box;
  color: #25272D;
  font-family: Motiva Sans;
  font-size: 0.875rem;
  line-height: 1.25;
  margin: 0.25rem 0;
  padding: 0.5rem;
  width: 100%;
}

@media (max-width:768px) {
  .emotion-6 {
    font-size: 1rem;
  }
}

.emotion-6::-webkit-input-placeholder {
  color: #989A9C;
}

.emotion-6:-moz-placeholder {
  color: #989A9C;
}

.emotion-6::-moz-placeholder {
  color: #989A9C;
}

.emotion-6:-ms-input-placeholder {
  color: #989A9C;
}

.emotion-6:focus {
  border: 1px solid #081D17;
  box-shadow: 0 0 8px rgba(182,239,208,0.1);
  outline: none;
}

.emotion-6:disabled {
  background: #F2F2F2;
}

.emotion-6:required {
  box-shadow: none;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 {
  color: #5B5D61;
  display: block;
  font-family: Motiva Sans;
  font-size: 0.875rem;
  line-height: 1.5;
}

<div
  className="emotion-8 emotion-9"
>
  <div
    className="emotion-4 emotion-3"
  >
    <div
      className="emotion-2 emotion-3"
    >
      <label
        className="emotion-0 emotion-1"
        htmlFor="example"
      >
        My Input
      </label>
    </div>
  </div>
  <input
    className="emotion-6 emotion-7"
    id="example"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
  
</div>
`;

exports[`<Input /> renders an Input with a type correctly 1`] = `
.emotion-4 {
  margin-bottom: 1rem;
}

.emotion-0 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  border-radius: 3px;
  border: 1px solid #E0E1E2;
  box-sizing: border-box;
  color: #25272D;
  font-family: Motiva Sans;
  font-size: 0.875rem;
  line-height: 1.25;
  margin: 0.25rem 0;
  padding: 0.5rem;
  width: 100%;
}

@media (max-width:768px) {
  .emotion-2 {
    font-size: 1rem;
  }
}

.emotion-2::-webkit-input-placeholder {
  color: #989A9C;
}

.emotion-2:-moz-placeholder {
  color: #989A9C;
}

.emotion-2::-moz-placeholder {
  color: #989A9C;
}

.emotion-2:-ms-input-placeholder {
  color: #989A9C;
}

.emotion-2:focus {
  border: 1px solid #081D17;
  box-shadow: 0 0 8px rgba(182,239,208,0.1);
  outline: none;
}

.emotion-2:disabled {
  background: #F2F2F2;
}

.emotion-2:required {
  box-shadow: none;
}

<div
  className="emotion-4 emotion-5"
>
  <div
    className="emotion-0 emotion-1"
  >
    
  </div>
  <input
    className="emotion-2 emotion-3"
    id="example"
    onBlur={[Function]}
    onChange={[Function]}
    type="number"
  />
  
</div>
`;

exports[`<Input /> renders as Input with a labelAside correctly 1`] = `
.emotion-6 {
  margin-bottom: 1rem;
}

.emotion-2 {
  font-family: Motiva Sans;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  border-radius: 3px;
  border: 1px solid #E0E1E2;
  box-sizing: border-box;
  color: #25272D;
  font-family: Motiva Sans;
  font-size: 0.875rem;
  line-height: 1.25;
  margin: 0.25rem 0;
  padding: 0.5rem;
  width: 100%;
}

@media (max-width:768px) {
  .emotion-4 {
    font-size: 1rem;
  }
}

.emotion-4::-webkit-input-placeholder {
  color: #989A9C;
}

.emotion-4:-moz-placeholder {
  color: #989A9C;
}

.emotion-4::-moz-placeholder {
  color: #989A9C;
}

.emotion-4:-ms-input-placeholder {
  color: #989A9C;
}

.emotion-4:focus {
  border: 1px solid #081D17;
  box-shadow: 0 0 8px rgba(182,239,208,0.1);
  outline: none;
}

.emotion-4:disabled {
  background: #F2F2F2;
}

.emotion-4:required {
  box-shadow: none;
}

.emotion-0 {
  font-family: Motiva Sans;
  color: #5B5D61;
  font-size: 0.875rem;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="emotion-6 emotion-7"
>
  <div
    className="emotion-2 emotion-1"
  >
    
    <div
      className="emotion-0 emotion-1"
      color="text.muted"
      fontSize="size4"
    >
      <div>
        sup?
      </div>
    </div>
  </div>
  <input
    className="emotion-4 emotion-5"
    id="example"
    onBlur={[Function]}
    onChange={[Function]}
    type="text"
  />
  
</div>
`;
